### PR TITLE
[FLINK-15793][k8s] Replace kubernetes-entry.sh with unified docker-entrypoint.sh

### PR DESCRIPTION
**Pull request by wangyang0918**
## What is the purpose of the change

Currently, the native K8s integration does not using the existing script(e.g. jobmanager.sh, taskmanager.sh) to start JobManager and TaskManager. Instead, the java start command is generate by Flink internally(i.e. `KubernetesClusterDescriptor`, `KubernetesResourceManager`).

In this PR https://github.com/apache/flink-docker/pull/37, we have integrated the functionality of `kubernetes-entry.sh` to the unified `docker-entrypoint.sh`.

This PR tries to remove the `kubernetes-entry.sh` completely.

**How to enable the plugin after this PR?**
```
flink run-application -t kubernetes-application -Dkubernetes.cluster-id=k8s-ha-app1 ...
-Dcontainerized.master.env.ENABLE_BUILT_IN_PLUGINS=flink-oss-fs-hadoop-1.11.1.jar \
-Dcontainerized.taskmanager.env.ENABLE_BUILT_IN_PLUGINS=flink-oss-fs-hadoop-1.11.1.jar \
local:///opt/flink/examples/streaming/StateMachineExample.jar
```

## Brief change log

* Replace kubernetes-entry.sh with unified docker-entrypoint.sh

## Verifying this change

* This PR is covered by existing unit tests(some tests need to be updated).
* The E2E tests(`test_kubernetes_session.sh, `test_kubernetes_application.sh`) should pass without modification.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / **don't** know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (**yes** / no / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangyang0918** [comment]
cc @rmetzger This PR is second part of FLINK-15793, which completely replace `kubernetes-entry.sh` with `docker-entrypoint.sh`. Please have a look at your convenience.

**Comment by wangyang0918** [comment]
@rmetzger Thanks for your suggestions. I have added two more commits to address your comments.
* de1e3d29fe97e5daecddebf4c8c705aba41f10db to update the documentation to show how to enable the plugins in native K8s integration
* 8c486a2bf8f9b604acddc6ea5b3d81413aa300b3 enrich the E2E to cover the plugin scenario. I think verifying the result is more reliable than just checking the logs.

File: docs/ops/deployment/native_kubernetes.md [hunk 1]
```diff
@@ -195,6 +188,28 @@ As always, Jobs may stop when manually canceled or, in the case of bounded Jobs,
...
+## Log Files
+
+By default, the JobManager and TaskManager will output the logs to the console and `/opt/flink/log` in each pod simultaneously.
+The STDOUT and STDERR will only be redirected to the console. You can access them via `kubectl logs <PodName>`.
+
+If the pod is running, you can also use `kubectl exec -it <PodName> bash` to tunnel in and view the logs or debug the process.
+
+## Using plugins
+
+In order to use [plugins]({{ site.baseurl }}/ops/plugins.html), they must be copied to the correct location in the Flink JobManager/TaskManager pod for them to work.
```
Comment (rmetzger) [inline docs/ops/deployment/native_kubernetes.md:201]:
```suggestion
In order to use [plugins]({{ site.baseurl }}/ops/plugins.html), they must be copied to the correct location in the Flink JobManager/TaskManager pod for them to work. You can use the built-in plugins without mounting a volume or building a custom Docker image.
```

File: docs/ops/deployment/native_kubernetes.md [hunk 2]
```diff
@@ -195,6 +188,28 @@ As always, Jobs may stop when manually canceled or, in the case of bounded Jobs,
...
+Use the following command to pass the environment variable to enable the S3 plugin for Flink application.
```
Comment (rmetzger) [inline docs/ops/deployment/native_kubernetes.md:202]:
```suggestion
For example, use the following command to pass the environment variable to enable the S3 plugin for your Flink application.
```

**Review by rmetzger — APPROVED** [review]
Thanks a lot! The E2E test looks nice!

I had a minor comment on the documentation, to make it more explicit that it is an example and that common filesystems work out of the box.

**Comment by wangyang0918** [comment]
@rmetzger Thanks for your nice text suggestion. I have integrated these comments and force pushed.

**Comment by rmetzger** [comment]
Thanks a lot! I'm merging the PR now!

**Pull request merged**